### ISSUE TITLE
Gene sequences

### ISSUE THREAD

**Issue by michael** [substance/texture#1102]
In JATS named content is used currently to identify field sequences (needs spec, provide examples).

@Melissa37 is this high priority, can you provide examples?

**Comment by Melissa37** [comment]
@michael we cannot go live without it. We don't know what papers will have gene sequences in them in advance of entering production, so we'd struggle to triage out papers in a phased launch.

The solution is very eLife specific, so I think a wider discussion would be useful, maybe with the JATS list/PMC? This tagging means our long genes do not fall off the webpage on the eLife site. I am not sure whether and how other publishers deal with this.

@FAtherden-eLife could you provide some samples?

**Comment by FAtherden-eLife** [comment]
@michael we use `<named-content content-type="sequence">` elements. We tend to only do it in the main flow of text - these sequences can occur in tables but, since this is only used for rendering purposes, it's typically not necessary to tag them.

#### Simple example in paragraph
This:
`<p>... (forward primer, <italic>Ov-</italic>GRN<italic>-</italic>1-RT-F: 5&#x0027;-GGGATCGGTTAGTCTAATCTCC and reverse primer, <italic>Ov-</italic>GRN1-RT-R: 5&#x0027;-GATCATGGGGGTTCACTGTC) ...</p>`

Is marked up as follows:
`<p>... (forward primer, <italic>Ov-</italic>GRN<italic>-</italic>1-RT-F: 5&#x0027;-<named-content content-type="sequence">GGGATCGGTTAGTCTAATCTCC</named-content> and reverse primer, <italic>Ov-</italic>GRN1-RT-R: 5&#x0027;-<named-content content-type="sequence">GATCATGGGGGTTCACTGTC</named-content>) ...</p>`

I think technically the actual primer sequence is the letters, that's why the `5'-` isn't within the `named-content` elements in the above, but since this isn't used for any semantic purpose for now, I don't think it would matter if it was included as well.


#### with formatting
It's also possible for authors to include other formatting in the sequences (for emphasis, or to distinguish certain strings inside the sequence), which _needs_ to be retained (our current proofing tool strips this formatting when sequences are tagged, which is problematic).

This example contains bold, underline, and colour (`<named-content content-type="author-callout-style-a.*">`) :
`<sec id="s4-2">
    <title>Vamp2 (1&#x2013;96, N29C)</title>
    <p>MSATAATVPPAAPAGEGGPPAPPPNLTS<bold><named-content content-type="author-callout-style-a3">C</named-content></bold>RRLQQTQAQVDEVVDIMRVNVDKVLE<named-content content-type="author-callout-style-a1">R</named-content>DQKLSELDDRADALQAGASQFETSAAKLKRKYWWKNLKMM<underline>GGSGNGSGGL<bold>C</bold>TPSRGGDYKDDDDK</underline></p>
</sec>`

And is marked up as follows:
`<sec id="s4-2">
    <title>Vamp2 (1&#x2013;96, N29C)</title>
    <p><named-content content-type="sequence">MSATAATVPPAAPAGEGGPPAPPPNLTS<bold><named-content content-type="author-callout-style-a3">C</named-content></bold>RRLQQTQAQVDEVVDIMRVNVDKVLE<named-content content-type="author-callout-style-a1">R</named-content>DQKLSELDDRADALQAGASQFETSAAKLKRKYWWKNLKMM<underline>GGSGNGSGGL<bold>C</bold>TPSRGGDYKDDDDK</underline></named-content></p>
</sec>`

Let me know if you need anything further.

**Comment by michael** [comment]
Thank you Fred, that's very helpful. I think a "Concepts" tagging feature in Texture could address this. I'd like to move a way from things like `author-callout-style-a3` and instead use a defined list of concept types that can be configured per publisher. If colours are really important, I'd add these to the configuration. Still I'd try to avoid colours in the editor as that's pretty arbitrary, you could still decide to render colors on the eLife page.

- gene-sequence (Gene Sequence, yellow)
- protein (Protein, green)
- organ (Organ, blue)

How does that approach sound to you?

**Comment by FAtherden-eLife** [comment]
Thanks @michael - I (personally) agree regarding the colours - I think we should be moving away from allowing colours in text and instead, rely on on other formatting (italic, bold underline etc.) for emphasis if it's needed. @Melissa37 would you agree? Are there any editorial implications in forcing authors to use other means of emphasis?

We (eLife) don't make a distinction for the different types of sequences since we don't use it for any semantic reason. It's purely used as a flag for the HTML output, as in - the content contained in this element can/should be broken across lines (word-break: break-all;).

So in our use case, we'd only need one way of tagging this type of content. But I can't speak for other publishers/use cases. As Melissa says above, perhaps a wider discussion would be of use.

**Comment by michael** [comment]
If the typing is not needed it could wait and we just define things as a "concept". Later if required one could optionally choose a type (e.g. gene sequence).

Can you point me to an article, where you have gene sequences so I can inspect the markup and style?

**Comment by FAtherden-eLife** [comment]
Sure, [this one](https://elifesciences.org/articles/41771/#s4-2) has sequences with all different kinds of formatting.

[This one](https://elifesciences.org/articles/42078#s4-1) is more typical of how they appear (inside a paragraph and with no emphasis).

Thanks

**Comment by Melissa37** [comment]
> I (personally) agree regarding the colours - I think we should be moving away from allowing colours in text and instead, rely on on other formatting (italic, bold underline etc.) for emphasis if it's needed. @Melissa37 would you agree? Are there any editorial implications in forcing authors to use other means of emphasis?

Yes I agree, I am adding this to the eLife trello board for Production process to do - I need to discuss and agree with the planning team

>If the typing is not needed it could wait and we just define things as a "concept". Later if required one could optionally choose a type (e.g. gene sequence).
@michael can you show us what you mean? Is it that there are concepts which are free-form, and then publishers define them and it outputs as required in their XML?

**Comment by oliver----** [comment]
@michael I think it would be cleaner to use the @vocab attributes to specify the concept. For Gene Sequences it could be @vocab-term-identifier='gene-sequence'. If there was an existing vocabulary for such concepts we could use it.

**Comment by michael** [comment]
Yeah we could consider that, and I checked those attributes too. Just wasn't sure if we want to go that far already. If we used the vocab attributes I'd expect to really link with an item in the database. While when we say this is a concept of category x, we are way less specific and it could be a first step before we introduce entity tagging that points to real resources in databases.

**Comment by oliver----** [comment]
Just to make clear, why I am against using @specific-use -- from the JATS documentation:

> Best Practice: This attribute is not used to make semantic distinctions

https://jats.nlm.nih.gov/archiving/tag-library/1.2d1/attribute/specific-use.html

Now only the question, how bad is it to use @vocab-term without specifying a real @vocab.

**Comment by michael** [comment]
Fair point, I think the semantic tagging use-cases must be discussed more before we act. So I suggest different phases. First phase are untyped concepts, to support a MVP for eLife, second phase would involve vocab-linking (to be further discussed).

I've added a section in the requirements document:

https://docs.google.com/document/d/1ZOjKrQZOndU9G12bVaXmFt56aWRW4kJSDjaMjcxSHSc/edit#heading=h.vqo846q59mon

**Comment by Melissa37** [comment]
Hi there

I am liking this! However, I would like to check in with the JATS community about it. Could you send an email to the JATS listserve with your proposal or ask Debbie?

It makes sense to have their experience weigh in on this too.

Melissa

**Comment by michael** [comment]
Debbie is pinged! :)

**Comment by mark-donoghue** [comment]
Received your post to the JATS mailing list.

@vocab seems to be the way to go. 

https://jats.nlm.nih.gov/articleauthoring/tag-library/1.2d2/attribute/vocab.html

This is a possible tagging of your example above:

```
<named-content
  content-type="concept"
  vocab="gene-sequence"
  vocab-identifier="https://genedb.org/genes#"
  vocab-term="Vamp2-1-96, N29C"
  vocab-term-identifier="http://genedb.org/genes#Vamp2-1-96,%32N29C">
MSATAATVPPAAPAGEGGPPAPPPNLTS<b>C</b>RRLQQTQAQVDEVVDIMRVNVDKVLE<bold>R</bold>DQKLSELDDRADALQAGASQFETSAAKLKRKYWWKNLKMM<italic>GGSGNGSGGL<bold>C</bold>TPSRGGDYKDDDDK</italic>
</named-content>
```

@vocab is uncontrolled so it can contain anything meaningful to your application.

**Comment by oliver----** [comment]
@mark-donoghue very helpful, thanks!
